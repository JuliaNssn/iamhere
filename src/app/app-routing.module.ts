import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CheckInComponent } from './check-in/check-in.component';
import { OverviewComponent } from './overview/overview.component';

const routes: Routes = [
  {
    path: '',
    component: OverviewComponent
  },
  {
    path: 'check-in',
    component: CheckInComponent
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}
