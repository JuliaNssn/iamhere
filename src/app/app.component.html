<header>
  <mat-toolbar>
    <span>Room.io</span>
  </mat-toolbar>
</header>
<main>
  <router-outlet></router-outlet>
</main>
