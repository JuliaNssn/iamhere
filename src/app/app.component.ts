import { Component } from '@angular/core';
import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';
import { State } from './shared/store/reducers';
import { loadUsers } from './shared/user/user.actions';
import { User } from './shared/user/user.model';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  users$: Observable<Array<User>> = this.store.select(
    (state: State) => state.user.users
  );

  constructor(private store: Store<State>) {
    this.store.dispatch(loadUsers());
  }
}
