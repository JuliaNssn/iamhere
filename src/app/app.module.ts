import { NgModule } from '@angular/core';
import { MatToolbarModule } from '@angular/material/toolbar';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { environment } from 'src/environments/environment';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { OverviewModule } from './overview/overview.module';
import { CheckinDataAccessModule } from './shared/checkin/data-access/checkin-data-access.module';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    CheckinDataAccessModule.forRoot(environment.apiUrl),
    AppRoutingModule,
    OverviewModule,
    BrowserAnimationsModule,
    MatToolbarModule
  ],
  bootstrap: [AppComponent]
})
export class AppModule {}
