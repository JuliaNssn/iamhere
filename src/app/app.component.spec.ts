import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { AppModule } from './app.module';
import { AppTestingModule } from './shared/testing/app-testing.module';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [AppModule, AppTestingModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
