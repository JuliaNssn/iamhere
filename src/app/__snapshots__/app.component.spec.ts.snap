// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snapshot 1`] = `
<app-root>
  <header>
    <mat-toolbar
      class="mat-toolbar"
    >
      <span>
        Room.io
      </span>
    </mat-toolbar>
  </header><main>
    <router-outlet />
  </main>
</app-root>
`;
